adoptingerlang.org

```diff
@@ -3444,44 +3446,74 @@ initContainers:
-
-Probes are defined either as a command to execute in the container, as an HTTP =GET= request made to a given port and path, or as an attempt to open a TCP connection to a specific port.
-
-A =livenessProbe= is optional; if the process in the container crashes or stops for any reason, it will be restarted. The probe ensures that if our node does not crash but for some reason has deadlocked and can not respond to requests, it will also be restarted.
-
-In =service_discovery= a =livenessProbe= is included that starts probing =/healthz= with an HTTP request after an initial delay of 3 seconds and then does so every 15 seconds.
-
-#+BEGIN_SRC yaml
-livenessProbe:
-  httpGet:
-    path: /healthz
-    port: http
-  initialDelaySeconds: 3
-  periodSeconds: 15
-#+END_SRC
-
-The =readinessProbe= is checked by Kubernetes to know if requests should be routed to a container. If the application needs to do some heavy lifting on startup, or periodically, where it won't be able to serve requests, it should return a =503= status response for readiness.
-
-Another use is to begin sending =503= status responses after receiving the =SIGTERM= signal from Kubernetes that tells the node to shutdown. It isn't required because Kubernetes will remove the =Pod= from the routes either way if it has sent it a shutdown signal, but this removal is not done before sending the signal, so for a period there will still be requests and readiness checks after the node has received the signal.
-
-The only necessary thing to do for graceful shutdown is to delay =init:stop()= from running until the =Pod= is removed from the =kube-proxy= =Endpoint=. The process can wait up to 30 seconds before Kubernetes sends a =SIGKILL= signal that will force the process to terminate.
-
-In =service_discovery= the =readinessProbe= is on a separate path, =/ready=, and with a higher frequency, every 5 seconds.
+**** Readiness, Liveness and Startup Probes
+
+Each container in a =Pod= can define a =readiness=, =liveness= and =startup= probe. The
+=liveness= probe . A startup probe can be added to delay when the =liveness= and
+=readiness= probes start being checked.
+
+Probes are defined either as a command to execute in the container, as an HTTP
+=GET= request made to a given port and path, or as an attempt to open a TCP
+connection to a specific port.
+
+A =livenessProbe= is not required to ensure restart of failed containers. If the
+process in the container crashes or stops for any reason, it will already be
+restarted. The probe should only be used for cases the process does not crash
+but for some reason has deadlocked and can not respond to requests. For this
```

Comment:
```suggestion
While kubernetes will restart any failed container, there are some times where a
process becomes unresponsive, such as deadlocks. A =livenessProbe= can be
used for such cases, where stopping to respond prompts a restart. For this
```
Trying to state what it is for, rather than what it isn't for first.